### PR TITLE
Add disclaimer for necessary fake models

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a disclaimer for the fake models we can't really remove since the canonical checkpoint is very big already. You can see the result in the [preview](https://moon-ci-docs.huggingface.co/docs/transformers/pr_21178/en/model_doc/gptj#transformers.GPTJModel.forward.example).

If this is acceptable, I'll add it to the other places where we want a tiny random model instead of the huge one for the docstrings.

### CODE REVIEW

File: src/transformers/utils/doc.py [hunk 1]
```diff
@@ -1118,6 +1131,11 @@ def docstring_decorator(fn):
         code_sample = filter_outputs_from_example(
             code_sample, expected_output=expected_output, expected_loss=expected_loss
         )
+        if real_checkpoint is not None:
+            intro = FAKE_MODEL_DISCLAIMER.format(
```
Comment (ydshieh) [inline src/transformers/utils/doc.py:1135]: Is it possible to not format `here`, but just simply `code_sample = intro + code_sample` directly, and let `built_doc = code_sample.format(**doc_kwargs)` at the end to handle the format just once?

File: src/transformers/utils/doc.py [hunk 2]
```diff
@@ -146,6 +146,17 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
 
 
+FAKE_MODEL_DISCLAIMER = """
+    <Tip warning={true}>
+
+    This example uses a random model as the real ones are all very big. To get proper results, you should use
+    {real_checkpoint} instead of {fake_checkpoint}. If you get out-of-memory when loading that checkpoint, you can try
+    adding `device_map="auto"` in the `from_pretrained` call.
+
+    </Tip>
```
Comment (ydshieh) [inline src/transformers/utils/doc.py:156]: Should there be an empty line after `</Tip>` - in order to distinguish with the actual code sample block?

**Review by ydshieh — APPROVED** [review]
It's definitely a big ➕ to have this clear message! Thank you! Just 1 nit for you to decide.

File: src/transformers/utils/doc.py (hunk 2 above)
Comment (LysandreJik) [inline src/transformers/utils/doc.py:156]: Yes, great!

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**